src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -239,6 +244,17 @@ class Queue[T <: Data](val gen: T,
     maybe_full := do_enq
```

Comment:
Seems there exists a unhandled double write behavior?
What if `do_enq =/= do_deq && io.flush.get`?
`enqueue`(but not `dequeue`) and `flush` at the same cycle.

Reply:
@sequencer can you re-review to see if your concern is now addressed?